src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -249,7 +245,7 @@ public override BoundNode VisitReturnStatement(BoundReturnStatement node)
                 var expression = node.ExpressionOpt;
                 if (expression != null)
                 {
-                    var returnType = expression.GetTypeAndNullability(_includeNullability);
+                    var returnType = expression.Type;
```

Comment:
It's not obvious to me why `_types` isn't a `TypeSymbolWithAnnotations` anymore.
It seems that flow analysis would need the information about top-level nullability too.

Reply:
`NullableWalker` should revisit the lambda body to determine the return expression nullability from flow analysis. In short, the nullability from initial binding will be ignored.

---
In reply to: [164234258](https://github.com/dotnet/roslyn/pull/23981#discussion_r164234258) [](ancestors = 164234258)